src/odf/dataset/src/services/metadata_chain_visitor.rs

```diff
@@ -94,6 +94,14 @@ pub trait MetadataChainVisitor: Send {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+pub trait ExtractBlock {
```

Comment:
With the approach used in `dataset_metadata_handler` we probably will not need this trait